atlasdb-remoting-api/src/main/java/com/palantir/atlasdb/config/AuxiliaryRemotingParameters.java

```diff
@@ -37,6 +37,17 @@
      */
     boolean shouldRetry();
 
+    /**
+     * Whether this client should support operations that block on the server side.
+     * In practice, this means that read and idle connection timeouts may be longer.
+     *
+     * This is set to true by default, to support legacy behaviour.
+     */
+    @Value.Default
+    default boolean shouldSupportBlockingOperations() {
```

Comment:
I suppose for the lock service, it's only the lock endpoint that you want to be able to block no?

You could create an impl of the lock service which uses the "supports blocking operations" for the lock method, and then the standard for everything else.

Reply:
oh, you did this in #4520 💃